planner/core/logical_plan_builder.go

```diff
@@ -5875,11 +5875,44 @@ func buildWindowSpecs(specs []ast.WindowSpec) (map[string]*ast.WindowSpec, error
 	return specsMap, nil
 }
 
+func unfoldSelectList(list *ast.SetOprSelectList, unfoldList *ast.SetOprSelectList) {
+	for _, sel := range list.Selects {
+		switch s := sel.(type) {
+		case *ast.SelectStmt:
+			unfoldList.Selects = append(unfoldList.Selects, s)
+		case *ast.SetOprSelectList:
+			unfoldSelectList(s, unfoldList)
+		}
+	}
+}
+
 // extractTableList extracts all the TableNames from node.
 // If asName is true, extract AsName prior to OrigName.
 // Privilege check should use OrigName, while expression may use AsName.
+// TODO: extracting all tables by vistor model maybe a better way
 func extractTableList(node ast.ResultSetNode, input []*ast.TableName, asName bool) []*ast.TableName {
 	switch x := node.(type) {
+	case *ast.SubqueryExpr:
+		input = extractTableList(x.Query, input, asName)
+	case *ast.SelectStmt:
+		input = extractTableList(x.From.TableRefs.Left, input, asName)
+		input = extractTableList(x.From.TableRefs.Right, input, asName)
+		switch w := x.Where.(type) {
```

Comment:
`From` and `Where` are covered, what about the scalar subquery (column subquery)?

Reply:
`column subquery` goes wrong, I will fix it later